### PR TITLE
fix CCFileUtils::getPathForFilename() for ios, mac / add Lua support

### PR DESCRIPTION
**Pull request by dualface**
fix: append missed "/" char
add: Lua support

### CODE REVIEW

**Comment by dumganhar** [comment]
Hi, @dualface,
We have refactored CCFileUtils for all platforms.
We have abstracted the same implementations of CCFileUtils.
So could you pull our latest codes and send it again?

File: cocos2dx/platform/CCFileUtilsCommon_cpp.h [hunk 1]
```diff
@@ -410,11 +410,21 @@ const std::vector<std::string>& CCFileUtils::getSearchResolutionsOrder()
...
+}
+
 const std::vector<std::string>& CCFileUtils::getSearchPaths()
 {
     return m_searchPathArray;
 }
 
+void CCFileUtils::addSearchPath(const char* path)
+{
+    m_searchPathArray.push_back(path);
```
Comment (dumganhar) [inline cocos2dx/platform/CCFileUtilsCommon_cpp.h:425]: You should add the default root resource path if developer passed in a relative path.
You could refer to `CCFileUtils::setSearchPaths`.

Comment (dualface) [inline cocos2dx/platform/CCFileUtilsCommon_cpp.h:425]: thanks, I will fix it.

**Comment by dualface** [comment]
> We have refactored CCFileUtils for all platforms.
> We have abstracted the same implementations of CCFileUtils.
> So could you pull our latest codes and send it again?

great jobs! I pull again when you ready.

**Comment by dumganhar** [comment]
The changes of refactoring CCFileUtils have been merged.

**Comment by dualface** [comment]
OK, i got.

**Comment by dualface** [comment]
See : https://github.com/cocos2d/cocos2d-x/pull/1953
This pull discarded.

**Pull request closed without merging**